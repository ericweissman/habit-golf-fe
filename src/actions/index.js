export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (error) => ({
  type: 'HAS_ERRORED',
  error
})

export const loginUserSuccess = (user) => ({
  type: 'LOGIN_USER_SUCCESS',
  user
})

export const setActiveClub = (club) => ({
  type: 'SET_ACTIVE_CLUB',
  club
})

export const setLastShot = (shot) => ({
  type: 'SET_LAST_SHOT',
  shot
})