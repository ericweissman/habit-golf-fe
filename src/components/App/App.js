import React, { Component } from 'react';
import Login from '../Login/Login'
import NewUser from '../NewUser/NewUser'
import Ratings from '../Ratings/Ratings'
import Home from '../Home/Home'
import './App.scss';

class App extends Component {
  state = {
    user: [1],
    filter: 'home'
  }

  changeFilter = (filter) => {
    this.setState({ filter })
  }

  componentToDisplay = () => {
    const { filter } = this.state
    switch (filter) {
      case 'ratings':
        return <Ratings changeFilter={this.changeFilter} />
      default:
        return <Home changeFilter={this.changeFilter}/>
    }
  }

  render() {
    const { user} = this.state
    return (
      <div className="App">
      <h1>habit golf</h1>
      { 
        user.length === 0 ? <Login></Login> : this.componentToDisplay()
      }
      </div>
    );
  }
}

export default App;
