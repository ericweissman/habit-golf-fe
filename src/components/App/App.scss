.App {
  background-color: #77D781;
  min-height: 1000px;
}
