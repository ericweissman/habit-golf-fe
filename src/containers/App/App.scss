.App {
  background-color: rgb(106, 176, 113);
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 700px;
  h1 {
    padding-top: 30px;
    font-size: 40px;
    font-weight: 200;
    color: #fffffffc;
  }
}

@media screen and (min-width: 359px) and (min-height: 740px) {
  .App {
    min-height: 750px;
    h1 {
      margin-top: 60px;
    }
  }
}

@media screen and (min-width: 375px) {
  .App {
    h1 {
      font-size: 44px;
    }
  }
}


@media screen and (min-width: 375px) and (min-height: 812px) {
  .App {
    min-height: 870px;
    min-width: 400px;
    h1 {
      margin-top: 50px;
    }
  }
}

@media screen and (min-width: 411px) and (min-height: 736px) {
  .App {
    min-height: 750px;
    h1 {
      font-size: 50px;
    }
  }
}

@media screen and (min-width: 411px) and (min-height: 823px) {
  .App {
    min-height: 870px;
    min-width: 400px;
    h1 {
      margin-top: 50px;
    }
  }

@media screen and (min-width: 414px) {
  .App {
    min-height: 739px;
    h1 {
      font-size: 50px;
    }
  }
}

@media screen and (min-width: 415px) {
  .App {
    min-height: 1000px;;
    h1 {
      font-size: 50px;
    }
  }
}

}