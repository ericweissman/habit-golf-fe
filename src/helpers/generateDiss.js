export const generateDiss = () => {
  const jabs = ['oof', 'woof', 'yeesh', 'yikes', 'yuck', 'not great', 'oh no', 'sheesh', 'hmmm']
  const feedback = jabs[Math.floor(Math.random() * jabs.length)]
  return feedback
}