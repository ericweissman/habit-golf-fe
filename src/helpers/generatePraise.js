export const generatePraise = () => {
  const praises = ['wow', 'impressive', 'great shot', 'superb', 'excellent', 'bravo']
  const feedback = praises[Math.floor(Math.random() * praises.length)]
  return feedback
}