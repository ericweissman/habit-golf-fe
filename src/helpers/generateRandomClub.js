export const generateRandomClub = () => {
  const clubs = ['wedges', 'irons', 'woods']
  const randomClub = clubs[Math.floor(Math.random() * clubs.length)]
  return randomClub
}
