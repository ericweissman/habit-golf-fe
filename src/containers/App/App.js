import React, { Component } from 'react';
import SignIn from '../../components/SignIn/SignIn'
import NewUser from '../../components/NewUser/NewUser'
import Error from '../../components/Error/Error'
import PremiumContent from '../../components/PremiumContent/PremiumContent'
import Ratings from '../Ratings/Ratings'
import Stats from '../Stats/Stats'
import Home from '../../components/Home/Home'
import { generateRandomClub } from '../../helpers/generateRandomClub'
import './App.scss';
import { connect } from 'react-redux';
import { setActiveClub } from '../../actions'

export class App extends Component {
  state = {
    user: [1],
    filter: 'home',
    newUser: false,
  }

  componentDidMount() {
    this.chooseRandomClub()
  }

  loginToDisplay = () => {
    const { newUser } = this.state
    if (!newUser) {
      return <SignIn newUser={this.toggleNewUser} />
    } else {
      return <NewUser />
    }
  }

  toggleNewUser = () => {
    this.setState({ newUser: !this.state.newUser })
  }


  chooseRandomClub = () => {
    const randomClub = generateRandomClub()
    this.props.setActiveClub(randomClub)
  }

  changeFilter = (filter) => {
    this.setState({ filter })
  }

  changeClub = (club) => {
    this.props.setActiveClub(club)
  }

  componentToDisplay = () => {
    const { filter } = this.state
    const { club } = this.props
    switch (filter) {
      case 'ratings':
        return <Ratings changeFilter={this.changeFilter} club={club} random={this.chooseRandomClub} />
      case 'stats':
        return <Stats changeFilter={this.changeFilter} />
      case 'paid':
        return <PremiumContent changeFilter={this.changeFilter} />
      default:
        return <Home changeFilter={this.changeFilter} club={club} changeClub={this.changeClub} />
    }
  }

  render() {
    const { user } = this.state
    const { error } = this.props
    return (
      <div className="App">
        <h1>habit golf</h1>
        {
          error && <Error error={error} />
        }
        {
          user.length === 0 ? this.loginToDisplay() : this.componentToDisplay()
        }
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
  error: state.error,
  club: state.club,
});


export const mapDispatchToProps = (dispatch) => ({
  setActiveClub: (club) => dispatch(setActiveClub(club))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
