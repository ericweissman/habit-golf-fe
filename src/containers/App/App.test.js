import React from 'react';
import { shallow } from 'enzyme'
import  { App, mapStateToProps, mapDispatchToProps } from '../App/App';
import { setActiveClub } from '../../actions'
import * as helper from '../../helpers/generateRandomClub'

helper.generateRandomClub = jest.fn(() => 'irons')

describe('App', () => {
  let wrapper
  const mockProps = {
    isLoading: false,
    error: '',
    club: 'wedges',
    setActiveClub: helper.generateRandomClub
  }
  
  const mockProps2 = {
    isLoading: false,
    error: 'ERROR',
    club: 'wedges',
    setActiveClub: helper.generateRandomClub
  }


  it('should correctly match the snapshot', () => {
    wrapper = shallow(<App {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the correct snapshot if the filter is Ratings', () => {
    wrapper.setState({ filter: 'ratings' })
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the correct snapshot if the filter is Premium Content', () => {
    wrapper.setState({ filter: 'paid' })
    expect(wrapper).toMatchSnapshot()
  })


  it('should correctly match the snapshot if there is no user logged in', () => {
    wrapper.setState({user: []})
    expect(wrapper).toMatchSnapshot()
  })

  it('should correctly match the snapshot if there is a new user', () => {
    wrapper.setState({newUser: true})
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the correct snapshot if there is an Error', () => {
    wrapper = shallow(<App {...mockProps2}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the proper default state', () => {
    wrapper = shallow(<App {...mockProps} />)
    const expectedState = {
      user: [1],
      filter: 'home',
      newUser: false,
    }
    expect(wrapper.state()).toEqual(expectedState)
  })


  it('should toggle the newUser in state when toggleNewUser is called', () => {
    const expectedState = {
      newUser: true,
      user: [1],
      filter: 'home',
    }
    wrapper.instance().toggleNewUser()
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should toggle filter when changeFilter is called', () => {
    const expectedState = {
      newUser: true,
      user: [1],
      filter: 'stats',
    }
    wrapper.instance().changeFilter('stats')
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should call setActiveClub when changeClub is called', () => {
    const mockClub = 'irons'
    wrapper.instance().changeClub(mockClub)
    expect(mockProps.setActiveClub).toHaveBeenCalled()
  })

  

  describe('mapStateToProps', () => {
    it('should return an object with isLoading, error, and club', () => {
      const mockState = {
        isLoading: false,
        error: '',
        club: 'wedges',
        fakeState: 1,
        fakeState2: 'a'
      }

      const expected = {
        isLoading: false,
        error: '',
        club: 'wedges',
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with the setActiveClub action', () => {
      const mockDispatch = jest.fn()
      const mockClub = 'irons'
      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = setActiveClub('irons')

      mappedProps.setActiveClub(mockClub)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})