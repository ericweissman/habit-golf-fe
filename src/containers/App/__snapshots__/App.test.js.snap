// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should correctly match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    club="wedges"
    error=""
    isLoading={false}
    setActiveClub={
      [MockFunction] {
        "calls": Array [
          Array [
            "woods",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          habit golf
        </h1>,
        "",
        <Home
          changeClub={[Function]}
          changeFilter={[Function]}
          club="wedges"
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "habit golf",
        },
        "ref": null,
        "rendered": "habit golf",
        "type": "h1",
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "changeClub": [Function],
          "changeFilter": [Function],
          "club": "wedges",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            habit golf
          </h1>,
          "",
          <Home
            changeClub={[Function]}
            changeFilter={[Function]}
            club="wedges"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "habit golf",
          },
          "ref": null,
          "rendered": "habit golf",
          "type": "h1",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changeClub": [Function],
            "changeFilter": [Function],
            "club": "wedges",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
